hera_cal/apply_cal.py

```diff
@@ -21,6 +23,17 @@ def recalibrate_in_place(data, data_flags, new_gains, cal_flags, old_gains=None,
         gain_convention: str, either 'divide' or 'multiply'. 'divide' means V_obs = gi gj* V_true,
             'multiply' means V_true = gi gj* V_obs. Assumed to be the same for new_gains and old_gains.
     '''
+    # get datatype of data_flags to determine if flags or wgts
+    if data_flags[data_flags.keys()[0]].dtype == np.bool:
+        bool_flags = True
+    elif data_flags[data_flags.keys()[0]].dtype == np.float:
+        bool_flags = False
+        wgts = data_flags
+        data_flags = DataContainer(dict(map(lambda k: (k, ~wgts[k].astype(np.bool)), wgts.keys())))
+    else:
+        raise ValueError("didn't recognize dtype of data_flags")
+
+    # loop over keys
     for (i,j,pol) in data.keys():
         if not np.all(data_flags[(i,j,pol)]):
```

Comment:
I think this line needs to get updated to handle weights. This might break if all the weights are zero because the visibility is totally flagged.